documentation/development/docker.md

```diff
@@ -0,0 +1,79 @@
+# Docker
+
+This describes how to create an environment for development using Docker.
+
+**This is still experimental, using [Vagrant](../installation.md) is much better tested and supported!**
+
+## 
+
+Add the following entries to your `.env` file:
+```dotenv
+ALLOWED_HOSTS='["*"]'
+DATABASE_URL='postgres://opencraft@postgresql/opencraft'
+DEFAULT_INSTANCE_MYSQL_URL='mysql://root@mysql'
+DEFAULT_RABBITMQ_API_URL='http://guest:guest@rabbitmq.local/api'
+DEFAULT_INSTANCE_RABBITMQ_URL='amqp://guest:guest@rabbitmq.local:5672/'
+DEFAULT_MONGO_REPLICA_SET_USER=''
+DEFAULT_MONGO_REPLICA_SET_PASSWORD=''
+DEFAULT_MONGO_REPLICA_SET_NAME='ocim'
+DEFAULT_MONGO_REPLICA_SET_PRIMARY='mongo'
+DEFAULT_MONGO_REPLICA_SET_HOSTS='mongo'
+REDIS_URL='redis://redis:6379'
+
+# not strictly necessary but useful to properly serve static assets
+# **in development**
+DEBUG=True
+```
+
+Start a one-off container for setup:
+```sh
+# --rm destroys the container at the end
+docker-compose run --rm ocim bash
+
+# then, inside the container...
+
+# initialize datastores
+make migrate
+# create super user
+make manage createsuperuser
+# exit the container
+exit
+```
+
+Then, start OCIM:
+```sh
+# DOCKER_PROFILES=ocim may be used instead of --profile
+# see https://docs.docker.com/compose/profiles/
+docker-compose --profile ocim up
+```
+
+The OCIM UI should be available at http://localhost:5000 .
+
+## Local installation with docker-compose services
+
+This quick setup uses Docker Compose to run the database services, while the Ocim instance runs in a regular virtualenv.
+
+Install the prerequisite packages:
+
+    sudo apt install make git libmysqlclient-dev mysql-client postgresql-client python3-dev python3-pip subversion unzip xvfb
```

Comment:
I'd prefer to standardize on the "run ocim in docker" approach, and not have this other approach which requires installing a lot of packages onto the host. Is there a compelling advantage to this approach?

Reply:
👍 I've deleted this section.